company_descriptions/compdescs_flow.py

```diff
@@ -0,0 +1,210 @@
+            frac=self.train_frac, random_state=69
+        )
+        train_indx = train.index
+
+        test = training_data_equal.drop(train_indx).drop(columns="job_id")
+
+        # convert to Dataset object
+        train_ds, test_ds = Dataset.from_pandas(train), Dataset.from_pandas(test)
+
+        self.ds = DatasetDict({"train": train_ds, "test": test_ds})
+
+        self.next(self.prepare_training_data)
+
+    @step
+    def prepare_training_data(self):
+        """Prepare training data by:
+        1) tokenizing
+        2) collating into batches
+        """
+        from transformers import DataCollatorWithPadding
+
+        self.tokenized_ds = self.ds.map(preprocess_function, batched=True)
+        self.data_collator = DataCollatorWithPadding(tokenizer=tokenizer)
+
+        self.next(self.train_model)
+
+    @batch(gpu=1, memory=60000, cpu=6, queue="job-queue-GPU-nesta-metaflow")
+    # @wandb_log(datasets=False, models=False, others=True)
+    @step
+    def train_model(self):
+        """
+        Fine tune the model on the labelled data
+        and evaluate it on a test set.
+        """
+        from transformers import TrainingArguments, Trainer
+        from scipy.special import softmax
+        import numpy as np
+        from sklearn.metrics import classification_report
+
+        self.learning_rate=2e-5
+        self.num_train_epochs=CONFIG["num_train_epochs"]
+        self.weight_decay=CONFIG["weight_decay"]
+        print("defining training arguments...")
+        self.output_dir = "jobbert-base-cased-compdecs"
+        training_args = TrainingArguments(
+            output_dir="jobbert-base-cased-compdecs",
+            learning_rate=self.learning_rate,
+            per_device_train_batch_size=CONFIG["per_device_train_batch_size"],
+            per_device_eval_batch_size=CONFIG["per_device_eval_batch_size"],
+            num_train_epochs=self.num_train_epochs,
+            weight_decay=self.weight_decay,
+        )
+        print("instantiating trainer...")
+        trainer = Trainer(
+            model=self.model,
+            args=training_args,
+            train_dataset=self.tokenized_ds["train"],
+            eval_dataset=self.tokenized_ds["test"],
+            tokenizer=self.tokenizer,
+            data_collator=self.data_collator,
+        )
+        print("training...")
+        trainer.train()
+
+        self.trainer_results = trainer.evaluate()
+        self.trainer_results["perplexity"] = round(exp(self.trainer_results["eval_loss"]), 2)
+
+        for config_name, config_value in CONFIG.items():
+            self.trainer_results[config_name] = config_value
+
+        # convert logits to probabilities
+        probs = softmax(trainer.predict(self.tokenized_ds["test"]).predictions, axis=1)
+        # convert probabilities to predicted labels
+        y_pred = np.argmax(probs, axis=-1)
+        # get true labels
+        y_true = self.tokenized_ds["test"]["label"]
```

Comment:
I might have missed something but should the test set be separate from the dataset used as `eval_dataset` at l.161? It looks like the same dataset is being used as part of evaluation during the fine-tuning process, and also to generate the final metrics. But I think there should be a separate hold-out test set, right?